.github/workflows/auto-cherry-picker.yaml

```diff
@@ -0,0 +1,47 @@
+name: Auto Cherry-Picker
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      PR_number:
+        description: The PR number to cherry-pick
+        type: string
+        required: true
+
+jobs:
+  inputs:
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    outputs:
+      merge_commit: ${{ steps.step1.outputs.merge_commit }}
+    steps:
+      - id: step1
+        name: Get merge commit
+        run: |
+          COMMIT=$(gh pr view --repo "${{ github.repository }}" "${{ github.event.pull_request.number || inputs.PR_number }}" --json mergeCommit --jq '.mergeCommit.oid')
+          echo "merge_commit=$COMMIT" >> "$GITHUB_OUTPUT"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  picker:
+    if: (github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'needs-cherrypick') ) || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    needs: inputs
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ needs.inputs.outputs.merge_commit }}
+
+      - name: Cherry-Pick
+        run: |
+          git config --local user.email "pantsbuild+github-automation@gmail.com"
+          git config --local user.name "Worker Pants (Pantsbuild GitHub Automation Bot) "
+          build-support/bin/cherry_pick.sh ${{ github.event.pull_request.number || inputs.PR_number }}
```

Comment:
Due to the `checkout`/`ref`, this will be running the build support script from the `merge_commit`, rather than the latest one on `main`. I suspect this is okay (they'll almost always be the same, except for in weird concurrent merge edge case), but just flagging that potential surprise.

Reply:
I'm not sure if that's true for merged PRs. But let's checkout `main` to be sure

Reply:
Hm, now that I think about it more, checking out `main` might be worse, because then the merge commit may not necessarily be available on the runner. So, either leave it as you had it, or _also_ fetch the merge commit explicitly.

(I also realised that the manual dispatch would be a non-weird case where the merge commit isn't the HEAD of `main`.)

Reply:
> or also fetch the merge commit explicitly.

Yeah that's the line you commented on where I was fetching from my fork. It's now doing the right thing and fetching from the base repo.